Remotion/ObjectBinding/Web.UnitTests/UI/Controls/BocAutoCompleteReferenceValueTest.cs

```diff
@@ -77,6 +77,7 @@ public override void SetUp ()
       MockRepository mockRepository = new MockRepository();
       _page = mockRepository.PartialMultiMock<Page> (typeof (ISmartPage));
       ((ISmartPage) _page).Stub (stub => stub.Context).Return (new HttpContextWrapper (HttpContext.Current));
+      ((ISmartPage) _page).Stub (stub => stub.Site).Return (null);

```

Comment:
Why is this necessary? Shouldn't be null the default for a partial mock even without stubbing?

Reply:
Rhino.Mocks behavior has changed between 3.6.0 & 3.6.1. If this stub is not set up, the following exception is thrown:
```
Rhino.Mocks.Exceptions.ExpectationViolationException : IComponent.get_Site(); Expected #0, Actual #1.
   at Rhino.Mocks.MethodRecorders.UnorderedMethodRecorder.DoGetRecordedExpectation(IInvocation invocation, Object proxy, MethodInfo method, Object[] args)
   at Rhino.Mocks.MethodRecorders.MethodRecorderBase.GetRecordedExpectation(IInvocation invocation, Object proxy, MethodInfo method, Object[] args)
   at Rhino.Mocks.Impl.ReplayPartialMockState.DoMethodCall(IInvocation invocation, MethodInfo method, Object[] args)
   at Rhino.Mocks.Impl.ReplayMockState.MethodCall(IInvocation invocation, MethodInfo method, Object[] args)
   at Rhino.Mocks.MockRepository.MethodCall(IInvocation invocation, Object proxy, MethodInfo method, Object[] args)
   at Rhino.Mocks.Impl.Invocation.Actions.RegularInvocation.PerformAgainst(IInvocation invocation)
   at Rhino.Mocks.Impl.RhinoInterceptor.Intercept(IInvocation invocation)
   at Castle.DynamicProxy.AbstractInvocation.Proceed()
   at Castle.Proxies.PageProxy.get_Site()
   at Remotion.Web.Utilities.ControlHelper.IsDesignMode(IControl control)
   at Remotion.ObjectBinding.Web.UI.Controls.BusinessObjectBinding.get_Property()
   at Remotion.ObjectBinding.Web.UI.Controls.BusinessObjectBoundWebControl.get_Property()
   at Remotion.ObjectBinding.Web.UI.Controls.BusinessObjectBoundEditableWebControl.get_IsReadOnly()
   at Remotion.ObjectBinding.Web.UI.Controls.BocAutoCompleteReferenceValue.GetTrackedClientIDs()
   at Remotion.ObjectBinding.Web.UnitTests.UI.Controls.BocAutoCompleteReferenceValueTest.GetTrackedClientIDsInEditMode()
```

Reply:
Interesting...